packages/core/src/connectors/coinbaseWallet.ts

```diff
@@ -57,8 +57,17 @@ export class CoinbaseWalletConnector extends Connector<
 
       const accounts = await provider.enable()
       const account = getAddress(accounts[0])
-      const id = await this.getChainId()
-      const unsupported = this.isChainUnsupported(id)
+
+      let id = await this.getChainId()
+      const unsupported = this.isChainUnsupported(chainId ?? id)
+      if (chainId && id !== chainId) {
+        const chain = await this.switchChain(chainId)
+        id = chain.id
+      } else if (unsupported) {
```

Comment:
It looks like `unsupported` will still be returned to the consumers as `true` once this branch is finished even though it's switched to a supported chain?

Reply:
ah yep, fixing.